src/Kupo/Data/Cardano/Script.hs

```diff
@@ -87,14 +88,30 @@ fromMaryScript =
     Ledger.Alonzo.TimelockScript . Ledger.Allegra.translateTimelock
 {-# INLINABLE fromMaryScript  #-}
 
-fromAlonzoScript
-    :: Ledger.Alonzo.Script (AlonzoEra StandardCrypto)
+-- TODO Remove unneeded unsafe upgrade from AlonzoEra to ConwayEra
+--  I could not find any evidence that we should upgrade
+--  from AlonzoEra to ConwayEra but to avoid introducing
+--  breaking changes I'm keeping current design through
+--  this unsafe function
```

Comment:
ConwayEra's script are a superset of AlonzoEra's scripts, like most types on the ledger; which is why Kupo has this approach of "upgrading" data into the most recent type. There's nothing _unsafe_ about it in principle.